### PR TITLE
adding nrpgx5 instrumentation tracer for github.com/jackc/pgx/v5

### PR DESCRIPTION
**Pull request by egon12**
This PR are adding instrumentation for nrpgx/v5 (https://github.com/jackc/pgx)

## Links

The discussion about new tracing in pgx https://github.com/jackc/pgx/issues/1061.
Open Telemetry Implementation for this tracing feature https://github.com/exaring/otelpgx/blob/main/tracer.go

## Details

The pgx library has new tracing support in v5. Before pgx only support for logging that only provide us with duration of the operation. With this new tracing feature, we can create new Segment at `QueryStart` and End the Segment with `QueryEnd`

For passing the segment between `QueryStart` and `QueryEnd` we use `context.Context`. This folllow the implementation from open telemetry.

And for the `Host`, `Port` and `Database Name`, I try to follow `database/sql` implementation in `nrpq` and `nrpgx` by using BaseSegment.

To use the library you can use this code
```go
cfg, _ := pgx.ParseConfig(context.Backgroud(), "<url here>")
cfg.Tracer = nrpgx5.NewTracer()

pgx.ConnectConfig(context.Background(), cfg)
```

### CODE REVIEW

**Comment by iamemilio** [comment]
Hi @egon12, thanks for contributing. Could you please include a simple example of how to use this library in a directory named `example/`.

**Comment by iamemilio** [comment]
I am not sure at which point in this extension we should be incorporating the go-agent

**Comment by egon12** [comment]
Just for info, this is related for this issues https://github.com/newrelic/go-agent/issues/386.
But In pgx/v4, we don't have tracer interface yet. So it's hard to add instrumentation. 
In pgx/v5, the maintainer add tracer interface. And we can use that to add newrelic instrumentation

**Comment by egon12** [comment]
Hello are there any update on this?

File: v3/integrations/nrpgx5/go.mod
Comment (iamemilio) [inline v3/integrations/nrpgx5/go.mod:9]:
```suggestion
	github.com/newrelic/go-agent/v3 v3.20.0
```

**Comment by iamemilio** [comment]
Looks good to me, thanks for submitting this. Once you address the comment I left, you have my approval!

**Review by iamemilio — APPROVED** [review]

**Comment by BertBR** [comment]
There's any equivalent implementation for v4?

**Pull request merged**
